LucianoConceicaoJunior/ss_back_trial
# Added Rspec and some adjustments

### What Was Changed

1. Added Rspec Tests: So that we can keep track in case any future change affects working features
2. Added API Key creation to user: Now right before a new user is created, its API Key will be generated
3. Adjusted Recommendation Engine: Added one more filter so that it won't recommend the movie that generated the list itself.
4. Removed duplication from GemFile: Just to clear it.

### How To Test

1. Run `bundle exec rspec -fd`. All Tests should pass
2. On Rails Console run `User.create! name: 'User1'` Then run `User.last.api_key`. API Key should be populated
3. Just fetch the Recommendations endpoint, it should work fine.
